rofi: migrate to rasi configuration format

thiagokokada opened the pull request:
### Description

<!--

Please provide a brief description of your change.

-->

The Xresources configuration format is deprecated in Rofi. For example, using Rofi from unstable (1.6.1 as of now) you get the following warnings when starting the application:

```
(process:9272): Rofi-WARNING **: 01:38:48.596: The old Xresources based configuration format is deprecated.

(process:9272): Rofi-WARNING **: 01:38:48.596: Please upgrade: rofi -upgrade-config.
``````

So this commit migrates it for its new configuration format, called rasi instead.

This new implementation uses attrsets manipulation instead of using strings, making the code clearer and also fixing some bugs found during the way. This is mostly an implementation detail though, since this change is mostly backwards compatible with the old one (the only option that is not backwards compatible is `programs.rofi.extraConfig`, but this would break anyway since the textual format changed). To make sure everything is right, I also created some tests.

If someone wants to validate if the generated config is correct, just run in terminal:

```
$ rofi -dump-config
```

And rofi will dump the current configuration file, including all unsetted options.

Closes issue #1593.

### Checklist

<!--

Please go through the following checklist before opening a non-WIP
pull-request.

Also make sure to read the guidelines found at

  https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-guidelines

-->

- [x] Change is backwards compatible. (mostly, except for `programs.rofi.extraConfig` that is documented)

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

thiagokokada commented on modules/programs/rofi.nix:
```diff
@@ -72,20 +72,14 @@ let
   };
 
   valueToString = value:
-    if isBool value then (if value then "true" else "else") else toString value;
```
This was a bug in the old code, when you set the value to `false` it wrote `else` to config instead.

thiagokokada commented:
Don't know what is failing in CI, both `rofi` related tests works in my machine.

berbiche commented:
> Don't know what is failing in CI, both `rofi` related tests works in my machine.

Yes, this is/was an issue on nixpkgs-unstable with kakoune due to a treewide refactor of stdenv.lib -> lib, the failing CI will only be resolved when the nixpkgs channel used in the CI is updated (which is automatic IIRC).

thiagokokada commented:
Removed the unnecessary blank lines from the result config file, also changed some function names to match their names in the `lib.generators`. I think this PR is ready to review.

thiagokokada commented:
Rebased with master and add myself as maintainer. I think now the tests should run.

berbiche commented on .github/CODEOWNERS:
```diff
@@ -89,7 +89,7 @@
 /modules/programs/mcfly.nix                           @marsam
 
-/modules/programs/mpv.nix                             @tadeokondrak
+/modules/programs/mpv.nix                             @tadeokondrak @thiagokokada
 
```
Missed this in #1726, you should also add `/tests/modules/programs/mpv`

thiagokokada replied:
Done.
